Bug: `test_no_detection_raises` reads global appdata cache → false negative on machines that have used the app

## Summary

`tests/unit/test_config.py::TestInstanceDetection::test_no_detection_raises` is environment-dependent and produces false negatives on any machine where the app has been launched at least once. The test passes a `tmp_path` for `cwd` but does not isolate the global `%APPDATA%/oar-priority-manager/last-instance.json` cache, which `detect_instance_root` reads as step 4 of its fallback chain. When the cache exists, step 4 returns a valid path and the expected `DetectionError` is never raised.

The bug is in the **test**, not the production code — the production fallback chain is correctly behaving as documented in the docstring at `src/oar_priority_manager/app/config.py:117`.

## Repro

On a machine where the app has been launched at least once (any `last-instance.json` exists at `%APPDATA%/oar-priority-manager/`):

```
QT_QPA_PLATFORM=offscreen .venv/Scripts/python.exe -m pytest tests/unit/test_config.py::TestInstanceDetection::test_no_detection_raises -v
```

```
FAILED tests/unit/test_config.py::TestInstanceDetection::test_no_detection_raises
E   Failed: DID NOT RAISE <class 'oar_priority_manager.app.config.DetectionError'>
```

On a fresh machine (no cache file) the test passes. **CI happens to pass because the GitHub Actions runner is fresh per-job and never accumulates a cache** — but any developer or contributor running the suite locally after using the app once will see this failure persistently.

## Root cause

`detect_instance_root` (`src/oar_priority_manager/app/config.py:106`) implements a 5-step fallback per spec §8.3.1:

1. `mods_path` arg
2. `cwd` contains MO2 markers
3. Walk up from `cwd`
4. **Cached `last-instance.json` from `%APPDATA%`** ← global state, not isolated by `tmp_path`
5. Raise `DetectionError`

The test passes only `cwd=tmp_path`. Steps 1–3 fail (correct). Step 4 reads `%APPDATA%/oar-priority-manager/last-instance.json` via `load_last_instance()` (line 154-156). If that file exists, the function returns its parent directory and never reaches step 5.

The test was likely written against an older mental model of the function before the cache fallback (step 4) was added — back when "no cwd match" meant "no detection." Now that step 4 exists, the test's premise is invalid on any machine with a populated cache.

## Fix direction (test-only change)

Pick one of these — they're roughly equivalent. Option A is the cleanest.

### A. Monkeypatch `load_last_instance` to return None

```python
def test_no_detection_raises(self, tmp_path: Path, monkeypatch):
    monkeypatch.setattr(
        "oar_priority_manager.app.config.load_last_instance",
        lambda: None,
    )
    with pytest.raises(DetectionError):
        detect_instance_root(cwd=tmp_path)
```

### B. Redirect APPDATA to `tmp_path` for the test

```python
def test_no_detection_raises(self, tmp_path: Path, monkeypatch):
    monkeypatch.setenv("APPDATA", str(tmp_path / "appdata"))
    with pytest.raises(DetectionError):
        detect_instance_root(cwd=tmp_path)
```

This depends on `_appdata_dir()` honoring `$APPDATA` (it does — line 33 of `config.py`).

### C. Apply the fix as a fixture

If other tests in `TestInstanceDetection` also benefit from cache isolation, use an autouse fixture at the class level rather than per-test patches.

## Acceptance criteria

- [ ] `tests/unit/test_config.py::TestInstanceDetection::test_no_detection_raises` passes regardless of whether `%APPDATA%/oar-priority-manager/last-instance.json` exists
- [ ] Other tests in `TestInstanceDetection` are audited for the same isolation gap (any test that calls `detect_instance_root()` may be affected — most pass `cwd` with valid markers and short-circuit at step 2, but verify)
- [ ] No production-code changes (the bug is in the test, not the function under test)
- [ ] Full pytest suite passes including the fixed test

## Out of scope

- Changes to `detect_instance_root`'s fallback chain — it behaves as specified
- Adding a CLI flag to disable the cache lookup — separate enhancement if ever needed

## Why this matters

This is a "passes in CI, fails on dev machines" bug — the worst kind for trust in the test suite. Anyone running `pytest` locally after launching the app once will see a persistent failure they can't trivially explain, which erodes the signal of a green test suite. Fixing this restores `pytest` as a deterministic local check.

🤖 _Generated by Claude Code on behalf of @cbeaulieu-gt_

--- a/tests/unit/test_config.py
+++ b/tests/unit/test_config.py
@@ -42,7 +42,19 @@ class TestInstanceDetection:
         result = detect_instance_root(cwd=nested)
         assert result == tmp_path
 
-    def test_no_detection_raises(self, tmp_path: Path):
+    def test_no_detection_raises(self, tmp_path: Path, monkeypatch):
+        """Step 4 (appdata cache) must be patched so the test is machine-independent.
+
+        Without the patch, a developer machine that has launched the app at
+        least once will have a valid last-instance.json in %APPDATA%, causing
+        step 4 to return a real path and preventing DetectionError from ever
+        being raised.  The monkeypatch forces load_last_instance to return
+        None, ensuring the full fallback chain exhausts at step 5.
+        """
+        monkeypatch.setattr(
+            "oar_priority_manager.app.config.load_last_instance",
+            lambda: None,
+        )
         with pytest.raises(DetectionError):
             detect_instance_root(cwd=tmp_path)
 

